internal/pkg/util/fs/fuse/fuse_mount_linux.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2023, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package fuse
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/sylabs/singularity/v4/internal/pkg/util/bin"
+	"github.com/sylabs/singularity/v4/pkg/image"
+	"github.com/sylabs/singularity/v4/pkg/sylog"
+)
+
+type ImageMount struct {
```

Comment:
I think this could become a more generic `fuse.Mount`. While it's an image here so far, that's not going to be generically the case.